### PR TITLE
[Torch Dialect] fix isValidSubtype with dynamic dim

### PR DESCRIPTION
**Pull request by qingyunqu**

### CODE REVIEW

File: test/Dialect/Torch/ops.mlir [hunk 1]
```diff
@@ -165,3 +165,8 @@ func.func @number_type_subtypes(%arg0: !torch.tensor, %arg1: !torch.list<int>, %
 
 func.func private @tensor_legal_dtype$torch.qint8() -> !torch.tensor<*,!torch.qint8>
 func.func private @tensor_legal_dtype$torch.quint8() -> !torch.tensor<*,!torch.quint8>
+
+func.func @prim_list_construct(%arg0: !torch.vtensor<[1,53,56,96],f16>, %arg1: !torch.vtensor<[1,3,56,96],f16>) -> !torch.list<vtensor<[1,?,56,96],f16>> {
+  %arg2 = "torch.prim.ListConstruct"(%arg0, %arg1) : (!torch.vtensor<[1,53,56,96],f16>, !torch.vtensor<[1,3,56,96],f16>) -> !torch.list<vtensor<[1,?,56,96],f16>>
+  return %arg2 : !torch.list<vtensor<[1,?,56,96],f16>>
+}
```
Comment (ramiro050) [inline test/Dialect/Torch/ops.mlir:172]: nit: call this something like `@prim_list_construct$valid_shape_subtype`

**Review by ramiro050 — APPROVED** [review]
Thanks!
